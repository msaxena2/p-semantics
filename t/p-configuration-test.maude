in "../p-semantics.maude" .

mod P-CONFIGURATION-TEST is including P-SEMANTICS .

  ops M1 M2 : -> Id .

endm

--- Soup
red <soup> < 1 | x = 1 | .Map > </soup> .

--- Multiple Cells
red <k> x = x + 1 ; </k>
    <machines>
      M1 |-> machine M1 ; { x = 2 ; }
      M2 |-> machine M2 ; { x = 1 ; }
    </machines> .


red <machines>
      M1 |-> machine M1 ; { x = 2 ; }
      M2 |-> machine M2 ; { x = 1 ; }
    </machines>
    <soup>
      < 0 | x : int ; y : int ; | .Map >
    </soup> .

quit .

