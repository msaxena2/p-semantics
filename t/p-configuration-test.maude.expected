==========================================
reduce in P-CONFIGURATION-TEST : 
<k>
  x = x + 1 ;
  y = y + 2 ;
</k> .
rewrites: 0
result Cell: 
<k>
  x = x + 1 ;
  y = y + 2 ;
</k>
==========================================
reduce in P-CONFIGURATION-TEST : 
<k>
  machine M1 ; {
    x = 2 ;
  }
</k> .
rewrites: 0
result Cell: 
<k>
  machine M1 ; {
    x = 2 ;
  }
</k>
==========================================
reduce in P-CONFIGURATION-TEST : 
<k>
  x = x + 1 ;
</k> 
<machines>
  
    M1 |-> machine M1 ; {
      x = 2 ;
    }
   
    M2 |-> machine M2 ; {
      x = 1 ;
    }
  
</machines> .
rewrites: 0
result Cells: 
<k>
  x = x + 1 ;
</k> 
<machines>
  
    M1 |-> machine M1 ; {
      x = 2 ;
    }
   
    M2 |-> machine M2 ; {
      x = 1 ;
    }
  
</machines>
==========================================
reduce in P-CONFIGURATION-TEST : 
<state>
  
    x |-> 23
   
    y |-> 4
  
</state> 
<machines>
  
    M1 |-> machine M1 ; {
      x = 2 ;
    }
   
    M2 |-> machine M2 ; {
      x = 1 ;
    }
  
</machines> .
rewrites: 0
result Cells: 
<state>
  
    x |-> 23
   
    y |-> 4
  
</state> 
<machines>
  
    M1 |-> machine M1 ; {
      x = 2 ;
    }
   
    M2 |-> machine M2 ; {
      x = 1 ;
    }
  
</machines>
Bye.
